pkg/database/service.go

```diff
@@ -266,7 +270,66 @@ func (s service) FindExternalDownload(uuid uuid.UUID) (model.ExternalDownload, e
 	return s.repository.FindExternalDownload(uuid)
 }
 
-func (s service) SaveAs(database *model.Database, instance *model.Instance, stack *model.Stack, newName string, format string) (*model.Database, error) {
+func (s service) Save(database *model.Database, instance *model.Instance, stack *model.Stack) error {
+	tmpName := randomString(20)
+	format := getFormat(database)
+	_, err := s.SaveAs(database, instance, stack, tmpName, format, func(saved *model.Database) {
+		err := s.Delete(database.ID)
+		if err != nil {
+			logError(err)
+			return
+		}
+
+		u, err := url.Parse(saved.Url)
+		if err != nil {
+			logError(err)
+			return
+		}
+
+		sourceKey := strings.TrimPrefix(u.Path, "/")
+		destinationKey := fmt.Sprintf("%s/%s", saved.GroupName, database.Name)
+		err = s.s3Client.Copy(s.c.Bucket, sourceKey, destinationKey)
+		if err != nil {
+			logError(err)
+			return
+		}
+
+		err = s.s3Client.Delete(s.c.Bucket, saved.Url)
+		if err != nil {
+			logError(err)
+			return
+		}
+
+		saved.Name = database.Name
+		saved.Url = database.Url
+		saved.Slug = database.Slug
+		err = s.Update(saved)
+		if err != nil {
+			logError(err)
+			return
+		}
+
+		err = s.repository.UpdateId(saved.ID, database.ID)
+		if err != nil {
+			logError(err)
+			return
+		}
+	})
+	if err != nil {
```

Comment:
please remove as this is already covered by the final return

Reply:
This handles any error potentially returned from the "save as"

Reply:
```go
if err != nil {
  return err
}

return err
```

should be

```go
return err
```

Reply:
Correct! I get it now, thanks :-)